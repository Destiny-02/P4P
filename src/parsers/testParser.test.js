const { join } = require("node:path");
const { readdirSync, promises: fs } = require("node:fs");
const { getParserForLanguage } = require("./getParserForLanguage");

const testCaseFolderPath = join(__dirname, "tests");
const testCaseFileNames = readdirSync(testCaseFolderPath);

// this file automatically creates test cases for every file in the "tests"
// directory. We shouldn't need to edit this file.

describe("test case files", () => {
  for (const fileName of testCaseFileNames) {
    const parser = getParserForLanguage(fileName);
    if (parser) {
      // generate a test case for each file

      it(`parses ${fileName} (${parser.language})`, async () => {
        const pathToFile = join(testCaseFolderPath, fileName);
        expect(await parser.parse(pathToFile)).toMatchSnapshot();
      });
    } else {
      it.todo(`parses ${fileName} (no parser available)`);
    }
  }
});
