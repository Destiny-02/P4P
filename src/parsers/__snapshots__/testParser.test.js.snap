// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test case files parses Sample.java (java) 1`] = `
{
  "comments": [
    "/**
   * Sample is a class
   */",
    "// this is a comment",
  ],
  "identifiers": [
    {
      "name": "parsers",
      "type": "packageDeclaration",
    },
    {
      "name": "languages",
      "type": "packageDeclaration",
    },
    {
      "name": "__tests__",
      "type": "packageDeclaration",
    },
    {
      "name": "input",
      "type": "packageDeclaration",
    },
    {
      "name": "Sample",
      "type": "typeIdentifier",
    },
    {
      "name": "get_annual_income_deduction",
      "type": "methodDeclarator",
    },
    {
      "name": "a",
      "type": "variableDeclaratorId",
    },
    {
      "name": "b",
      "type": "variableDeclaratorId",
    },
    {
      "name": "a",
      "type": "fqnOrRefTypePartCommon",
    },
    {
      "name": "b",
      "type": "fqnOrRefTypePartCommon",
    },
    {
      "name": "paye_tax_bracket",
      "type": "variableDeclaratorId",
    },
    {
      "name": "String",
      "type": "unannClassType",
    },
    {
      "name": "get_payslip_for_paye",
      "type": "methodDeclarator",
    },
  ],
}
`;

exports[`test case files parses sample.js (javascript) 1`] = `
{
  "comments": [
    "*
 * comments yay
 * @param {number} a
 * @param {number} b
 ",
    " an inline comment",
  ],
  "identifiers": [
    {
      "name": "a",
      "type": "FunctionDeclaration.argument",
    },
    {
      "name": "b",
      "type": "FunctionDeclaration.argument",
    },
    {
      "name": "add",
      "type": "FunctionDeclaration.ownId",
    },
  ],
}
`;

exports[`test case files parses sample.py (python) 1`] = `
{
  "comments": [
    "
Hiii
",
    "# inline commenttt",
  ],
  "identifiers": [
    {
      "name": "x",
      "type": "FunctionDeclaration.argument",
    },
    {
      "name": "y",
      "type": "FunctionDeclaration.argument",
    },
    {
      "name": "subtract",
      "type": "FunctionDeclaration.ownId",
    },
  ],
}
`;
